public/components/register_model/tag_field.tsx

```diff
@@ -112,31 +140,29 @@ export const ModelTagField = ({ index, tagKeys, tagValues, onDelete }: ModelTagF
 
   const onKeyCreate = useCallback(
     (value: string) => {
-      if (value.length > MAX_TAG_LENGTH) {
-        return;
-      }
       tagKeyController.field.onChange(value);
     },
     [tagKeyController.field]
   );
 
   const onValueCreate = useCallback(
     (value: string) => {
-      if (value.length > MAX_TAG_LENGTH) {
-        return;
-      }
       tagValueController.field.onChange(value);
     },
     [tagValueController.field]
   );
 
   const keyOptions = useMemo(() => {
-    return tagKeys.map((key) => ({ label: key }));
-  }, [tagKeys]);
+    return tagKeys
+      .filter((key) => !tags?.find((tag) => tag.key === key))
+      .map((key) => ({ label: key }));
+  }, [tagKeys, tags]);
 
   const valueOptions = useMemo(() => {
-    return tagValues.map((value) => ({ label: value }));
-  }, [tagValues]);
+    return tagValues
```

Comment:
It feels counter-intuitive that tag value is exclusive, for example, the tag key `Accuracy: test` and `Accuracy: validation` may have the same value: `0.95`

Reply:
You're right. We don't need to remove the selected value option. If we use the explicit type tag design, the number value will be a number input. The value options list should from the previous tag values for a specific key.